pkg/client/cache/reflector_test.go

```diff
@@ -360,3 +360,43 @@ func TestReflector_ListAndWatchWithErrors(t *testing.T) {
 		r.ListAndWatch(util.NeverStop)
 	}
 }
+
+func TestReflectorResync(t *testing.T) {
```

Comment:
I thought about it, but I didn't come up with anything reasonable.
The only way to do it that came to my mind is some kind of integration test with modified Transport layer performing some checks. I will try to do something like that in a separate PR.